shared-ts/jsonTypes.ts

```diff
@@ -0,0 +1,113 @@
+
+enum SourceType {
+    directReport,
+    email,
+    phone,
+    ax,
+    post,
+    feed,
+    radio,
+    tv,
+    web,
+    pager,
+    text,
+    other,
+}
+
+enum Progress {
+    draft, 
+    pendingApproval, 
+    approvedDraft, 
+    open, 
+    closing, 
+    closed, 
+    rejected
+}
+
+enum Reason {
+    Accident,
+    SecurityAlert,
+    Congestion,
+    RoadWorks,
+    RoadClosed,
+    BombExplosion,
+    Incident,
+    RouteDiversion,
+    StaffAbsence,
+    SignalProblem,
+    MaintenanceWork,
+    ConstructionWork,
+    Fog,
+    Ice,
+    HeavyRain,
+    WaterLogged,
+    Unknown
+}
+
+enum DayType {
+    monday,
+    tuesday,
+    wednesday,
+    thursday,
+    friday,
+    saturday,
+    sunday
+}
+
+export type Source = {
+    SourceType: SourceType;
+    TimeOfCommunication: string;
+};
+
+export type Period = {
+    StartTime: Date;
+    EndTime?: Date;
+};
+
+export type InfoLink = {
+    Uri: string;
+}
+
+export type Consequences = {
+    Condition: string;
+}
+
+export type Repetitions = {
+    DayType: DayType;
+}
+
+export type Reference = {
+    RelatedToRef: string[];
+};
+
+export type PtSituationElement = {
+    CreationTime: Date;
+    ParticipantRef: string;
+    SituationNumber: string;
+    Version?: Number;
+    Source: Source;
+    Progress: Progress;
+    ValidityPeriod: Period[];
+    PublicationWindow: Period;
+    MiscellaneousReason?: Reason;
+    Planned: boolean;
+    Summary: string;
+    InfoLink?: InfoLink[];
+    Consequences: Consequences[];
+    Repetitions?: Repetitions[];
```

Comment:
I think Repetitions isn't an array here but DayType should be, it also needs to come after ValidityPeriod

Reply:
Updated the Daytype to an array